packages/wdio-globals/src/index.ts

```diff
@@ -0,0 +1,82 @@
+type SupportedGlobals = 'browser' | 'driver' | 'multiremotebrowser' | '$' | '$$'
+
+const globals: Map<SupportedGlobals, any> = new Map()
+const GLOBALS_ERROR_MESSAGE = 'Don\'t import @wdio/globals outside of the WDIO testrunner context'
+
+type BrowserSync = import('webdriverio').Browser<'async'>
+type ElementSync = import('webdriverio').Element<'async'>
+type MultiRemoteBrowserAsync = import('webdriverio').MultiRemoteBrowser<'async'>
+type ElementArrayImport = import('webdriverio').ElementArray
+
+declare namespace WebdriverIOAsync {
+    interface Browser {}
+    interface Element {}
+    interface MultiRemoteBrowser {}
+}
+
+declare namespace WebdriverIO {
+    interface Browser extends BrowserSync, WebdriverIOAsync.Browser { }
+    interface Element extends ElementSync, WebdriverIOAsync.Element { }
+    // @ts-expect-error
+    interface MultiRemoteBrowser extends MultiRemoteBrowserAsync, WebdriverIOAsync.MultiRemoteBrowser { }
+    interface ElementArray extends ElementArrayImport {}
+}
+
+declare global {
+    var browser: WebdriverIO.Browser
+    var driver: WebdriverIO.Browser
+    var multiremotebrowser: WebdriverIO.MultiRemoteBrowser
+    var $: (...args: Parameters<WebdriverIO.Browser['$']>) => ReturnType<WebdriverIO.Browser['$']>
+    var $$: (...args: Parameters<WebdriverIO.Browser['$$']>) => ReturnType<WebdriverIO.Browser['$$']>
+}
+
+function proxyHandler (key: SupportedGlobals) {
+    return {
+        get: (self: never, prop: any) => {
+            if (!globals.has(key)) {
+                throw new Error(GLOBALS_ERROR_MESSAGE)
+            }
+
+            return globals.get(key)[prop]
+        }
+    }
+}
+
+export const browser = new Proxy(
+    class Browser {} as any as WebdriverIO.Browser,
```

Comment:
I think we can do the following instead
```suggestion
    class Browser extends WebdriverIO.Browser {},
```
This also goes for the similar instances below.

Reply:
I don't think this works given that `WebdriverIO.Browser` is a type and not a class.